src/nanothread.cpp

```diff
@@ -63,6 +65,65 @@ struct Worker {
 
 static Pool *pool_default_inst = nullptr;
 static std::mutex pool_default_lock;
+static uint32_t cached_core_count = 0;
+
+uint32_t core_count() {
+    // assumes atomic word size memory access
+    if (cached_core_count)
+        return cached_core_count;
+
+#if defined(__linux__)
+    /* Determine the number of present cores */
+    uint32_t ncores = sysconf(_SC_NPROCESSORS_CONF);
```

Comment:
Is that the same as `std::thread::hardware_concurrency`?

Reply:
I think so, again sysconf is the legacy way of getting this information it seems.